add OpenBSD 6.2

trombik opened the pull request:
this PR add OpenBSD 6.2.

also:

* OpenBSD boxes fetche packages from one of the official CDN
* install `bsd.mp`, fixes #55 

as a result, all OpenBSD boxes will be updated

Commit:
switch to CDN mirror and prepare 6.2

Commit:
[bugfix] s/&&/-a/

Commit:
missing backticks

Commit:
add OpenBSD 6.2

mitsururike commented on scripts/openbsd/init.sh:
```diff
@@ -3,9 +3,12 @@
 
-sudo tee /etc/pkg.conf <<EOF
-installpath = ftp.openbsd.org
+# pkg.conf has been replaces with installurl since 6.1
```
typo: s/replaces/replaced/

mitsururike commented on scripts/openbsd/init.sh:
```diff
@@ -42,7 +45,7 @@ fi
 # package builder first. remove this when the builder is ready.
 #
-if [ `ansible --version | head -n 1 | cut -d' ' -f 2` == '2.3.2.0' ]; then
+if [ `ansible --version | head -n 1 | cut -d' ' -f 2` == '2.3.2.0' -a `uname -r` != '6.2' ]; then
```
ansible package is installed from our ports tree on `6.2` and the patch is necessary.
or exclude `6.2` before the line 15 `sudo pkg_delete ansible`?

mitsururike commented on scripts/openbsd/init.sh:
```diff
@@ -42,7 +45,7 @@ fi
 # package builder first. remove this when the builder is ready.
 #
-if [ `ansible --version | head -n 1 | cut -d' ' -f 2` == '2.3.2.0' ]; then
+if [ `ansible --version | head -n 1 | cut -d' ' -f 2` == '2.3.2.0' -a `uname -r` != '6.2' ]; then
```
our ports tree has been patched for `6.2`?

trombik replied:
have not looked at the `patches` directory yet, but the package in `OPENBSD_6_2` has passed specs that check the bug in question.

mitsururike commented on reallyenglish_spec/packages_spec.rb:
```diff
@@ -15,24 +15,27 @@
 when "openbsd"
   # XXX RE_5_9 does not have the latest ansible yet
-  if os[:release].to_f >= 6.0
+  if os[:release].to_f >= 6.0 && os[:release].to_f < 6.2
```
i do not understand why 6.2 and above are excluded.

Commit:
typo

Commit:
typo

https://github.com/reallyenglish/packer-templates/pull/54#discussion_r145627470

Commit:
refactor the logic to patch ansible

of course, it is still messy. it will be fixed once we have package
builders and ansible ports are updated to the same one.

also:

* fix a typo, `uanme`

trombik commented:
all boxes have passed the tests.

Commit:
run boxes with two CPUs

this installs bsd.mp on OpenBSD

Commit:
test if bsd.sp is there instead of sysctl

what is tested here is the MP kernel has been installed

trombik commented:
all boxes have passed the tests.

mitsururike reviewed (commented):
tests in a box booted with multiple CPUs should be added to ensure that the installed kernel can boot single and multi CPU box and uses all the CPUs.

trombik commented:
i have been working on another feature branch. the tests you have mentioned would be implemented in the branch

trombik commented:
f71aa2c adds the tests
